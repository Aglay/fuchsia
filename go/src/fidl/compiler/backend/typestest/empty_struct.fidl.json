{
  "version": "0.0.1",
  "name": "test.name",
  "library_dependencies": [],
  "const_declarations": [],
  "enum_declarations": [],
  "interface_declarations": [],
  "struct_declarations": [
    {
      "name": "test.name/Empty",
      "anonymous": false,
      "members": [],
      "size": 1,
      "alignment": 1,
      "max_handles": 0
    }
  ],
  "table_declarations": [],
  "union_declarations": [],
  "declaration_order": [
    "test.name/Empty"
  ],
  "declarations": {
    "test.name/Empty": "struct"
  }
}
