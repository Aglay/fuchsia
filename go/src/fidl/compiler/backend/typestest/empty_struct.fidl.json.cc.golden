// WARNING: This file is machine generated by fidlgen.

#include <empty_struct.fidl.json.h>

#include "lib/fidl/cpp/internal/implementation.h"
namespace test {
namespace name {

extern "C" const fidl_type_t test_name_EmptyTable;
const fidl_type_t* Empty::FidlType = &test_name_EmptyTable;

void Empty::Encode(::fidl::Encoder* encoder, size_t offset) {
  ::fidl::Encode(encoder, &__reserved, offset + 0);
}

void Empty::Decode(::fidl::Decoder* decoder, Empty* value, size_t offset) {
  ::fidl::Decode(decoder, &value->__reserved, offset + 0);
}

zx_status_t Empty::Clone(Empty* result) const {
  zx_status_t _status = ::fidl::Clone(__reserved, &result->__reserved);
  if (_status != ZX_OK)
    return _status;
  return ZX_OK;
}

bool operator==(const Empty& lhs, const Empty& rhs) {
  if (!::fidl::Equals(lhs.__reserved, rhs.__reserved)) {
    return false;
  }
  return true;
}
}  // namespace name
}  // namespace test
